sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/DistributeConstants.java

```diff
@@ -201,6 +216,16 @@
      */
     static final String PREFERENCE_KEY_POSTPONE_TIME = PREFERENCE_PREFIX + "postpone_time";
 
+    /**
+     * Preference key for update setup failure package hash.
+     */
+    static final String PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY = PREFERENCE_PREFIX + "update_setup_failed_package_hash";
+
+    /**
+     * Preference key for update setup failure error message.
+     */
+    static final String PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY = PREFERENCE_PREFIX + "update_setup_failed_message";
```

Comment:
Do we really need this one? I saw the same implementation in iOS but it will be removed soon.

Reply:
I believe we need this to store/retrieve error message. What could be the alternative approach for storing the message? Should we use instance variable for this?

Reply:
Yup, I just removed it on the iOS one, I saved it but never actually used it. The reason I didn't need to use the saved one was because I called the function to show the error dialog in the same scope as the function that parses the query parameters in the redirect URL, so it still had access to that message in a variable.

Reply:
I don't think this comment applies to the Android implementation, you have to persist the message to avoid losing it after kill/restart.

Reply:
Do we care of losing the message after kill/restart in this case? This step is very beginning and we can simply start over the process. It is not kinda handling download or update.

Reply:
We care about not opening browser twice, that would defeat the purpose I think.